spark/src/main/resources/error/delta-error-classes.json

```diff
@@ -1127,7 +1127,7 @@
   },
   "DELTA_INVALID_PROTOCOL_VERSION" : {
```

Comment:
I just noticed that incompatible table features go through `DeltaTableFeatureException` instead of `InvalidProtocolVersionException`. As a follow-up, it might be worth doing this same exercise for that exception type. 

WARNING: It looks like enriching table feature exceptions could be a bigger effort: There are a _lot_ of different table feature exception classes, which means a lot of different call sites. Hopefully the existing four enrichment points would catch most of those (if the helper just started wrapping the additional exception), but I suspect we might need to "enrich" an unknown number of additional code sites to catch everything.

Recommend filing an issue to track this, unless you are excited to tackle the PR yourself right away.

Reply:
I split the existing issue into two: https://github.com/delta-io/delta/issues/2082, https://github.com/delta-io/delta/issues/2215